10_msi_prediction/nbconverted/lasso_range.py

```diff
@@ -173,37 +178,69 @@
 # In[8]:
 
 
-# try with a float-valued x-axis
-# this is probably more "correct" than treating each lasso parameter as a
-# category (above plot); here the spaces between parameters reflect their
-# actual real-valued distance in log-space
-sns.set({'figure.figsize': (12, 5)})
-sns.set_style('ticks')
-
 plot_df = (
     perf_df[(perf_df.signal == 'signal')]
+      .merge(nz_coefs_df, left_on=['holdout_cancer_type', 'lasso_param', 'seed', 'fold'],
+             right_on=['cancer_type', 'lasso_param', 'seed', 'fold'])
+      .drop(columns=['cancer_type'])
       .sort_values(by=['holdout_cancer_type', 'lasso_param'])
       .reset_index(drop=True)
 )
 plot_df.lasso_param = plot_df.lasso_param.astype(float)
+plot_df.head()
+
+
+# In[9]:
+
+
+sns.histplot(plot_df.nz_coefs)
+for q in np.linspace(0.1, 0.9, 9):
+    print(f'{q:.3f}', f'{plot_df.nz_coefs.quantile(q):3f}')
+    plt.gca().axvline(x=plot_df.nz_coefs.quantile(q), color='black', linestyle=':')
+
+
+# In[10]:
+
+
+plot_df['nz_quantile'] = pd.qcut(
+    plot_df.nz_coefs,
+    q=np.linspace(0, 1, 11),
+    labels=[f'{q}' for q in range(1, 11)]
+)
+
+print(plot_df.nz_quantile.unique())
+plot_df.head()
+
+
+# In[11]:
+
+
+# try with a float-valued x-axis
```

Comment:
This comment appears to be in here twice. Do you need it in both places?

Reply:
Nope! Good catch.